src/features/error/index.tsx

```diff
@@ -0,0 +1,87 @@
+import { Clock } from '@/assets/icons'
+import { HttpError, type ErrorResponse } from '@/lib/http'
+import { useNavigate, useRouteError, useSearchParams } from 'react-router-dom'
+import type { ErrorType } from '@/features/error/type'
+
+const normalizeErrorType = (type: string | null): ErrorType | null => {
+  if (type === 'token') return 'token'
+  if (type === 'notfound') return 'notfound'
+  if (type === 'badrequest') return 'badrequest'
+  if (type === 'unexpected') return 'unexpected'
+  if (type === 'invalidticket') return 'invalidticket'
+  if (type === 'emptytoken') return 'emptytoken'
+  return null
+}
+
+const getTypeFromStatus = (status: number | undefined): ErrorType | null => {
+  if (status === 410) return 'token'
+  if (status === 404) return 'notfound'
+  if (status === 400) return 'badrequest'
+  return null
+}
+
+const getDefaultMessage = (type: ErrorType) => {
+  switch (type) {
+    case 'token':
+      return '토큰이 만료되었습니다.'
+    case 'notfound':
+      return '요청하신 정보를 찾을 수 없습니다.'
+    case 'badrequest':
+      return '요청을 처리할 수 없습니다.'
+    case 'unexpected':
+      return '예상치 못한 오류가 발생했습니다.'
+    case 'invalidticket':
+      return '티켓을 찾을 수 없습니다.'
+    case 'emptytoken':
+      return '토큰을 찾을 수 없습니다.'
+  }
+}
+
+const ErrorPage = () => {
+  const navigate = useNavigate()
+  const routeError = useRouteError()
+  const [searchParams] = useSearchParams()
+  const typeFromQuery = normalizeErrorType(searchParams.get('type'))
+  const messageFromQuery = searchParams.get('message') || undefined
+
+  const typeFromRouteError =
+    routeError instanceof HttpError
+      ? getTypeFromStatus(routeError.status)
+      : getTypeFromStatus(
+          (routeError as { status?: number } | undefined)?.status
+        )
```

Comment:
HttpError형이 아닌 값이라면 예측 불가능한 Type Assertion으로 굳이 getTypeFromStatus를 호출하게 하지 않고, 차라리 `unknown`이라는 ErrorType을 추가하고 예외처리를 하는게 좋아보입니다.

```suggestion
      : 'unknown'
```